standard/classes.md

```diff
@@ -3480,8 +3480,8 @@ A set accessor corresponds to a method with a single value parameter of the prop
 Based on the presence or absence of the get and set accessors, a property is classified as follows:
 
 - A property that includes both a get accessor and a set accessor is said to be a ***read-write property***.
-- A property that has only a get accessor is said to be a ***read-only property***. It is a compile-time error for a read-only property to be the target of an assignment.
-- A property that has only a set accessor is said to be a ***write-only property***. Except as the target of an assignment, it is a compile-time error to reference a write-only property in an expression.
+- A property that has only a get accessor is said to be a ***read-only property***. It is a compile-time error for a read-only property to be the target of an assignment unless the property is ref-valued and returns a writeable reference.
+- A property that has only a set accessor is said to be a ***write-only property***. Except as the target of an assignment or as an argument to the `nameof` operator ([§12.8.23](expressions.md#12823-the-nameof-operator)), it is a compile-time error to reference a write-only property in an expression.
```

Comment:
```suggestion
- A property that has only a set accessor is said to be a ***write-only property***. Except as the target of an assignment or as the operand of the `nameof` operator ([§12.8.23](expressions.md#12823-the-nameof-operator)), it is a compile-time error to reference a write-only property in an expression.
```

I'd initially used "within" instead of "as", but it's invalid to further dereference the write-only property - so `nameof(Test.X.Y)` is invalid if `X` is a write-only property, but `nameof(Test.X)` is okay.